test/ketu/async/integration_test.clj

```diff
@@ -120,6 +120,8 @@
       (u/try-put! producer-chan ["1" resolve1])
       ;; Close the sink to make sure messages were sent
       (async/close! producer-chan)
+      ;; Allow the worker thread to consume all messages before aborting input
+      (Thread/sleep 500)
```

Comment:
Shouldn't be needed if `sink/stop!` keeps the original order of invocations.

Reply:
Indeed. Removed.